File: algorithms/javascript/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  let arr = []
  for (const [key, num] of nums.entries()) {
    if (arr[target - num] > -1) {
      return [arr[target - num], key]
    }
    arr[num] = key
  }
};

console.log(twoSum([2, 7, 11, 15], 9))

File: algorithms/python/two-sum.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

class Solution(object):
    def twoSum(self, nums, target):
        arr = {}
        for key, num in enumerate(nums):
          if (target - num) in arr:
            return [arr[target - num], key]
          arr[num] = key

print Solution().twoSum([2,7,11,15], 9)
